providers/keycloak/tests/unit/keycloak/auth_manager/test_keycloak_auth_manager.py

```diff
@@ -59,3 +88,228 @@ def test_serialize_user(self, auth_manager):
+            ],
+            [
+                "is_authorized_connection",
+                "DELETE",
+                ConnectionDetails(conn_id="test"),
+                "Connection#DELETE",
+                "Connection:test#DELETE",
+            ],
+            ["is_authorized_connection", "GET", None, "Connection#GET", None],
+            ["is_authorized_backfill", "POST", BackfillDetails(id=1), "Backfill#POST", "Backfill:1#POST"],
+            ["is_authorized_backfill", "GET", None, "Backfill#GET", None],
+            ["is_authorized_asset", "GET", AssetDetails(id="test"), "Asset#GET", "Asset:test#GET"],
+            ["is_authorized_asset", "GET", None, "Asset#GET", None],
+            [
+                "is_authorized_asset_alias",
+                "GET",
+                AssetAliasDetails(id="test"),
+                "AssetAlias#GET",
+                "AssetAlias:test#GET",
+            ],
+            ["is_authorized_asset_alias", "GET", None, "AssetAlias#GET", None],
+            [
+                "is_authorized_variable",
+                "PUT",
+                VariableDetails(key="test"),
+                "Variable#PUT",
+                "Variable:test#PUT",
+            ],
+            ["is_authorized_variable", "GET", None, "Variable#GET", None],
+            ["is_authorized_pool", "POST", PoolDetails(name="test"), "Pool#POST", "Pool:test#POST"],
+            ["is_authorized_pool", "GET", None, "Pool#GET", None],
+        ],
+    )
+    @pytest.mark.parametrize(
+        "status_code_type, status_code_resource, expected_one_call, expected_two_calls",
+        [
+            [200, 200, True, True],
+            [200, 403, True, True],
+            [403, 200, False, True],
+            [403, 403, False, False],
+        ],
+    )
+    @patch("airflow.providers.keycloak.auth_manager.keycloak_auth_manager.requests")
+    def test_is_authorized(
+        self,
+        mock_requests,
+        function,
+        method,
+        details,
+        permission_type,
+        permission_resource,
+        status_code_type,
+        status_code_resource,
+        expected_one_call,
+        expected_two_calls,
+        auth_manager,
+        user,
+    ):
+        expected_num_calls = 1 if status_code_type == 200 or not permission_resource else 2
+        resp1 = Mock()
+        resp1.status_code = status_code_type
+        resp2 = Mock()
+        resp2.status_code = status_code_resource
+
+        mock_requests.post.side_effect = [resp1, resp2]
+
+        result = getattr(auth_manager, function)(method=method, user=user, details=details)
+
+        token_url = "server_url/realms/realm/protocol/openid-connect/token"
+        payload = {
+            "grant_type": "urn:ietf:params:oauth:grant-type:uma-ticket",
+            "audience": "client_id",
+            "permission": permission_type,
+        }
+        headers = {
+            "Authorization": "Bearer access_token",
+            "Content-Type": "application/x-www-form-urlencoded",
```

Comment:
Nit: Maybe make those strings constants that can be defined and used in src and imported into test so that they're always in sync and we don't have many magic strings in this module?